File: 1988.Minimize Maximum Pair Sum in Array/solution.py
class Solution(object):
    def minPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()

        l = 0
        r = len(nums)-1
        max_p = 0
        while(l<r):
            curr = nums[l]+nums[r]
            if(curr>max_p):
                max_p = curr
            
            l+=1
            r-=1
        return max_p

File: 1988.Minimize Maximum Pair Sum in Array/solution.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int l = 0;
        int r = nums.size()-1;
        int curr = 0;
        int max_p = 0;
        while(l<r){
            curr = nums[l]+nums[r];
            max_p = max(max_p, curr);
            l++;
            r--;
        }
        return max_p;
    }
};